tests/integration/test_dataplex_integration_performance.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+
+import logging
+import shutil
+import sys
+
+import click.testing
+import pytest
+
+from clouddq.main import main
+from clouddq.utils import working_directory
+
+
+logger = logging.getLogger(__name__)
+
+@pytest.mark.dataplex
+class TestDataplexPerformanceIntegration:
+
+    @pytest.fixture(scope="session")
+    def runner(self):
+        return click.testing.CliRunner()
+
+    def test_cli_16_rb_10_rules(
+        self,
+        runner,
+        tmp_path,
+        gcp_project_id,
+        gcp_dataplex_region,
+        gcp_dataplex_lake_name,
+        gcp_dataplex_zone_id,
+        gcp_bq_dataset,
+        target_bq_result_dataset_name,
+        target_bq_result_table_name):
+        try:
+            temp_dir = Path(tmp_path).joinpath("clouddq_test_cli_16_rb_10_rules")
+            temp_dir.mkdir(parents=True)
+            source_configs = Path("tests").joinpath("resources", "configs_16_rb_10_rules.yml").absolute()
```

Comment:
Just wondering, is there any upper limit of num of different bindings (as combination of entity_column_filter, etc)? In other words, how many different tables/views are allowed to be unioned in the final step?

Reply:
there is an upper limit on the final union step. I think in the orders of 100s of aggregate table, each may contain ~50 rules. So the current upper limit is roughly ~5000 rules.

Reply:
ok, thanks for the clarification!